### PR TITLE
[cli] enhance buffer info

### PR DESCRIPTION
**Pull request by bukepo**
This commit enhances buffer info when using heap for message buffers.

### CODE REVIEW

File: src/core/api/message_api.cpp [hunk 1]
```diff
@@ -174,6 +174,7 @@ void otMessageGetBufferInfo(otInstance *aInstance, otBufferInfo *aBufferInfo)
     uint16_t  messages, buffers;
     Instance &instance = *static_cast<Instance *>(aInstance);
 
+    aBufferInfo->mTotalBuffers = instance.Get<MessagePool>().GetTotalBufferCount();
     aBufferInfo->mTotalBuffers = OPENTHREAD_CONFIG_NUM_MESSAGE_BUFFERS;
```
Comment (jwhui) [inline src/core/api/message_api.cpp:178]:
```suggestion
    aBufferInfo->mTotalBuffers = instance.Get<MessagePool>().GetTotalBufferCount();
```

File: src/core/common/message.hpp [hunk 2]
```diff
@@ -1468,6 +1468,13 @@ class MessagePool : public InstanceLocator, private NonCopyable
      */
     uint16_t GetFreeBufferCount(void) const;
 
+    /**
+     * This method returns the total number of buffers.
+     *
+     * @returns The total number of buffers.
+     */
```
Comment (simonlingoogle) [inline src/core/common/message.hpp:1475]:
```suggestion
     * @returns The total number of buffers.
     *
     */
```

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

with a nit

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**